py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/config_rt_vehicle.py

```diff
@@ -11,47 +11,95 @@ def config_type(self) -> ConfigType:
+                ('start_date', pyarrow.string()),
+                ('start_time', pyarrow.string()),
+                ('trip_id', pyarrow.string()), # actual label: id
+                # entity -> vehicle -> vehicle
+                ('vehicle_id', pyarrow.string()), # actual label: id
+                ('vehicle_label', pyarrow.string()), # actual label: label
+                ('vehicle_consist', pyarrow.list_(
+                                        pyarrow.struct(
+                                            [pyarrow.field('label',pyarrow.string()),]
+                                        )
+                                    )), # actual label: consist
             ])
     
     def record_from_entity(self, entity: dict) -> dict:
-        vehicle = entity['vehicle']
-        record = {
-            'vehicle_timestamp': vehicle.get('timestamp'),
-            'vehicle_id': entity.get('id'),
-            'vehicle_label': vehicle["vehicle"].get('label'),
-            'current_status': vehicle.get('current_status'),
-            'current_stop_sequence': vehicle.get("current_stop_sequence"),
-            'stop_id': vehicle.get("stop_id"),
-            'position': vehicle.get('position'),
-            'trip': vehicle.get("trip"),
-            'consist_labels': [],
+        transform_schema = {
+            'entity': (
+                ('id','entity_id'),
+            ),
+            'entity,vehicle': (
+                ('current_status',),
+                ('current_stop_sequence',),
+                ('occupancy_percentage',),
+                ('occupancy_status',),
+                ('stop_id',),
+                ('timestamp','vehicle_timestamp'),
+            ),
+            'entity,vehicle,position': (
+                ('bearing',),
+                ('latitude',),
+                ('longitude',),
+                ('speed',),
+            ),
+            'entity,vehicle,trip': (
+                ('direction_id',),
+                ('route_id',),
+                ('schedule_relationship',),
+                ('start_date',),
+                ('start_time',),
+                ('id','trip_id'),
+            ),
+            'entity,vehicle,vehicle': (
+                ('id','vehicle_id',),
+                ('label','vehicle_label',),
+                ('consist','vehicle_consist',),
+            )
         }
-        if vehicle['vehicle'].get('consist') is not None:
-            record['consist_labels'] = [consist["label"]
-                                        for consist
-                                        in vehicle["vehicle"].get("consist")]
+
+        def drill_entity(f):
+            ret_dict = entity
+            f_list = f.split(',')
+            if len(f_list) == 1:
+                return ret_dict
+            for k in f_list[1:]:
+                if ret_dict.get(k) is None:
+                    return None
+                ret_dict = ret_dict.get(k)
+            return ret_dict
```

Comment:
Little function to parse dictionary transform object, not sure if this is the best approach, but we'll need something to handle expected entity keys not being present.